burn-core/src/nn/rnn/lstm.rs

```diff
@@ -349,4 +349,103 @@ mod tests {
         assert_eq!(cell_state.shape().dims, [8, 10, 1024]);
         assert_eq!(hidden_state.shape().dims, [8, 10, 1024]);
     }
+
+    #[test]
+    fn test_batched_backward_pass() {
```

Comment:
I think we should cleanup that test, like a pretty standard backward pass test, without having to use the cat function. Just checking if the gradients are calculated can be enough here.